### ISSUE TITLE
support `failures_before_warning` on Consul checks

### ISSUE THREAD

**Issue by 116davinder** [hashicorp/nomad#19028]
### Nomad version
1.6.3

### Consul Version
1.16.3

### Operating system and Environment details
N/A

### Issues
1. Nomad doesn't allow consul UDP check to be configured aka ` type  = "udp"`

```hcl
.....
      service {
        name     = "kerberos-kdc-primary"
        port     = "kpropd"
        provider = "consul"

        check {
          type     = "udp"
          port     = "kdc"
          interval = "10s"
          timeout  = "2s"
          success_before_passing   = "1"
          failures_before_critical = "3"
        }
      }
....
```
```bash
* Check service: "kerberos-kdc-primary" check invalid: invalid check type ("udp"), must be one of grpc, tcp, http, script
```

2. Nomad doesn't allow consul warning status to be configured aka `failures_before_warning`

```bash
kerberos.nomad:649,11-34: Unsupported argument; An argument named "failures_before_warning" is not expected here.
```
```hcl
      service {
        name     = "kerberos-kpropd"
        tags     = ["kerberos", "kpropd"]
        port     = "kpropd"
        provider = "consul"

        check {
          type     = "tcp"
          port     = "kpropd"
          interval = "10s"
          timeout  = "2s"
          success_before_passing   = "1"
          failures_before_critical = "3"
          failures_before_warning  = "2"
        }
      }
```

### Reproduction steps

Just set above values in job and it fail as mentioned in the above cases.

#### Expected Result

1. consul doing UDP check
2. consul show warning status in health checks when registered from nomad.

#### Actual Result

N/A
### Job file (if appropriate)
N/A
### Nomad Server logs (if appropriate)
N/A
### References
Consul Checks: https://developer.hashicorp.com/consul/docs/services/configuration/checks-configuration-reference

**Comment by tgross** [comment]
Hi @116davinder! The limitation on health check types is documented in the [`service.check`](https://developer.hashicorp.com/nomad/docs/job-specification/service#check) docs:
> The Consul integration supports the `grpc`, `http`, `script`, and `tcp` checks.

And `failures_before_warning` is also not documented as an available option in the [`check` block docs](https://developer.hashicorp.com/nomad/docs/job-specification/check).

But I'll mark this as a feature request for roadmapping.

**Comment by tgross** [comment]
Actually, the request for `udp` would be a duplicate of https://github.com/hashicorp/nomad/issues/14094 so I'm going to rename this to specifically cover the `failures_before_warning` option.

**Comment by 116davinder** [comment]
yes, i wanted to make it as feature request as well but realized that after creating bug ticket and then unable to change label for it.
thanks for making changes.

**Comment by mikenomitch** [comment]
Closed by [#19336](https://github.com/hashicorp/nomad/pull/19336)
